Integrations/integration-email-sender.yml

```diff
@@ -0,0 +1,349 @@
+
+  Implemented in Python to support extensions.
+configuration:
+- display: Mail server hostname or IP address
+  name: host
+  defaultvalue: ""
+  type: 0
+  required: true
+- display: SMTP Port
+  name: port
+  defaultvalue: "25"
+  type: 0
+  required: true
+- display: Credentials
+  name: credentials
+  defaultvalue: ""
+  type: 9
+  required: false
+- display: Sender address
+  name: from
+  defaultvalue: noreply@demisto.com
+  type: 0
+  required: true
+- display: Use TLS for connection
+  name: tls
+  defaultvalue: "false"
+  type: 8
+  required: false
+- display: Do not validate server certificate (insecure)
+  name: insecure
+  defaultvalue: "false"
+  type: 8
+  required: false
+script:
+  script: |
+    from email.mime.audio import MIMEAudio
+    from email.mime.base import MIMEBase
+    from email.mime.image import MIMEImage
+    from email.mime.multipart import MIMEMultipart
+    from email.mime.text import MIMEText
+    from smtplib import SMTP
+    import base64
+    import mimetypes
+    from email import encoders
+    import re
+    import random
+    import string
+
+
+    # Following methods raise exceptions so no need to check for return codes
+    SERVER = SMTP(demisto.getParam('host'), int(demisto.getParam('port')))
+    SERVER.ehlo()
+    # TODO - support for non-valid certs
+    if demisto.getParam('tls'):
+        SERVER.starttls()
+    if demisto.getParam('credentials') and demisto.getParam('credentials')['identifier'] and demisto.getParam('credentials')['password']:
+        SERVER.login(demisto.getParam('credentials')['identifier'], demisto.getParam('credentials')['password'])
+    FROM = demisto.getParam('from')
+
+
+    def randomword(length):
+        """
+        Generate a random string of given length
+        """
+        letters = string.ascii_lowercase
+        return ''.join(random.choice(letters) for i in range(length))
+
+
+    def return_error(data):
+        """
+        Return error as result and exit
+        """
+        SERVER.quit()
+        demisto.results({
+            'Type': entryTypes['error'],
+            'ContentsFormat': formats['text'],
+            'Contents': data
+        })
+        sys.exit(0)
+
+
+    def guess_type(filename):
+        """
+        Return the maintype and subtype guessed based on the extension
+        """
+        ctype, encoding = mimetypes.guess_type(filename)
+        if ctype is None or encoding is not None:
+            # No guess could be made, or the file is encoded (compressed), so
+            # use a generic bag-of-bits type.
+            ctype = 'application/octet-stream'
+        return ctype.split('/', 1)
+
+
+    def handle_file(msg, filename, maintype, subtype, cid, data):
+        """
+        Add the attachment to the message and add the relevant header
+        """
+        if maintype == 'text':
+            # UTF-8 is a pretty safe bet
+            att = MIMEText(data, subtype, 'utf-8')
```

Comment:
We had problems with utf-8 in python multiple times
Do we want to make it configurable

Reply:
UTF-8 is the simplest but will need to test. I don't want to make it even more complicated for the user.